analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeFile.cs

```diff
@@ -0,0 +1,107 @@
+
+    [DataTestMethod]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Test, false)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Test, true)]
+    public void ShouldAnalyzeFile_TestExclusions_ReturnExpected(string filePath, string[] testExclusions, ProjectType projectType, bool shouldAnalyze) =>
+        AssertShouldAnalyzeFile(filePath, projectType, shouldAnalyze, testExclusions: testExclusions);
+
+    [DataTestMethod]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Test, false)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Test, true)]
+    public void ShouldAnalyzeFile_GlobalTestExclusions_ReturnExpected(string filePath, string[] globalTestExclusions, ProjectType projectType, bool shouldAnalyze) =>
+        AssertShouldAnalyzeFile(filePath, projectType, shouldAnalyze, globalTestExclusions: globalTestExclusions);
+
+    [DataTestMethod]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Product, false)]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Test, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Test, true)]
+    public void ShouldAnalyzeFile_Inclusions_ReturnExpected(string filePath, string[] inclusions, ProjectType projectType, bool shouldAnalyze) =>
+        AssertShouldAnalyzeFile(filePath, projectType, shouldAnalyze, inclusions: inclusions);
+
+    [DataTestMethod]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "Foo" }, ProjectType.Test, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Test, false)]
+    public void ShouldAnalyzeFile_TestInclusions_ReturnExpected(string filePath, string[] testInclusions, ProjectType projectType, bool shouldAnalyze) =>
+        AssertShouldAnalyzeFile(filePath, projectType, shouldAnalyze, testInclusions: testInclusions);
+
+    [DataTestMethod]
+    [DataRow("Foo", new string[] { "Foo" }, new string[] { "Foo" }, ProjectType.Product, false)]
+    [DataRow("Foo", new string[] { "NotFoo" }, new string[] { "Foo" }, ProjectType.Product, false)]
+    [DataRow("Foo", new string[] { "Foo" }, new string[] { "NotFoo" }, ProjectType.Product, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, new string[] { "NotFoo" }, ProjectType.Product, false)]
+    public void ShouldAnalyzeFile_MixedInput_ReturnExpected(string filePath, string[] inclusions, string[] exclusions, ProjectType projectType, bool shouldAnalyze) =>
```

Comment:
Can we do the same for the test `ProjectType.Test`, of course with the project type parameters.

Reply:
I think you have missed this one.